apps/managedidentity/managedidentity_test.go

```diff
@@ -50,112 +61,345 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
+	// todo update as required
+	testCases := []struct {
+		name           string
+		source         Source
+		endpoint       string
+		expectedSource Source
+		miType         ID
+	}{
+		{name: "testAzureArcSystemAssigned", source: AzureArc, endpoint: imdsDefaultEndpoint, expectedSource: AzureArc, miType: SystemAssigned()},
+		{name: "testAzureArcUserClientAssigned", source: AzureArc, endpoint: imdsDefaultEndpoint, expectedSource: AzureArc, miType: UserAssignedClientID("clientId")},
+		{name: "testAzureArcUserResourceAssigned", source: AzureArc, endpoint: imdsDefaultEndpoint, expectedSource: AzureArc, miType: UserAssignedResourceID("resourceId")},
+		{name: "testAzureArcUserObjectAssigned", source: AzureArc, endpoint: imdsDefaultEndpoint, expectedSource: AzureArc, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImds", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsClientAssigned", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedClientID("clientId")},
+		{name: "testDefaultToImdsResourceAssigned", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedResourceID("resourceId")},
+		{name: "testDefaultToImdsObjectAssigned", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImdsEmptyEndpoint", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsLinux", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsEmptyEndpointLinux", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
 	}
 
 	for _, testCase := range testCases {
-		t.Run(http.StatusText(testCase.code), func(t *testing.T) {
-			fakeErrorClient := mock.Client{}
-			responseBody, err := makeResponseWithErrorData(testCase.err, testCase.desc)
+		t.Run(string(testCase.source), func(t *testing.T) {
+			unsetEnvVars(t)
+			setEnvVars(t, testCase.source)
+			setCustomAzureArcFilePath(t, fakeAzureArcFilePath)
+
+			actualSource, err := GetSource(testCase.miType)
 			if err != nil {
-				t.Fatalf("error while forming json response : %s", err.Error())
+				t.Fatalf("error while getting source: %s", err.Error())
 			}
-			fakeErrorClient.AppendResponse(mock.WithHTTPStatusCode(testCase.code),
-				mock.WithBody(responseBody))
-			client, err := New(SystemAssigned(), WithHTTPClient(&fakeErrorClient))
+
+			if actualSource != testCase.expectedSource {
+				t.Errorf(errorExpectedButGot, testCase.expectedSource, actualSource)
+			}
+		})
+	}
+}
+
+func TestAzureArcReturnsWhenHimdsFound(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("Skipping test on macOS as HIMDS is not supported")
+	}
```

Comment:
Actually this test cannot run on my machine, I have a Mac for work. 
When I run all tests this will fails when it does not support Mac
thats why this condition.

Reply:
How does it fail?